go/enclave/rpc/vk_utils.go

```diff
@@ -0,0 +1,124 @@
+package rpc
+
+import (
+	"encoding/json"
+	"fmt"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ten-protocol/go-ten/go/common"
+	"github.com/ten-protocol/go-ten/go/enclave/vkhandler"
+	"github.com/ten-protocol/go-ten/go/responses"
+)
+
+// UserRPCRequest1 - decoded RPC argument accompanied by a logical sender
+type UserRPCRequest1[P any] struct {
```

Comment:
if you need 3, you can create a dedicated struct, and use the "1" again.

The 2 is really a convenience to avoid the need to create yet another struct.